### DISCUSSION TITLE
gpu branch installation problem

### DISCUSSION THREAD

**Discussion by cywhale** [biigle/core#482] [Q&A] [answered]
I have merged gpu branch and build ok by following official instructions. The problems come from docker-compose.yaml and artisan.
This version of docker-compose.yaml remove version: "2.3", but it would be regarded as "invalid" docker-compose.yaml when using ./artisan down
But if I add version: "2.3", there is an unsupported config option for "depoly" error .
if using version: "3", unsupported config option for "init" and "volumes_from". 
So, how could I solve this problem? thanks.

And in this upgrade, image: postgres:10-alpine move to image: postgres:14-alpine
As I try ./artisan migrate, it cause errors as following. Is there any administration steps I should do/fix for this migration? thanks a lot!

```
In Connection.php line 759:
                                                                                                
  SQLSTATE[08006] [7] could not translate host name "database" to address: Try again (SQL: sel  
  ect * from information_schema.tables where table_catalog = biigle and table_schema = public   
  and table_name = migrations and table_type = 'BASE TABLE')                                    
                                                                                                

In Connector.php line 70:
                                                                                      
  SQLSTATE[08006] [7] could not translate host name "database" to address: Try again   
```

**Comment by mzur** [comment, marked as answer]
Which Docker Compose version do you use? As far as I know the `version` is [deprecated](https://docs.docker.com/compose/compose-file/#compose-file).

> And in this upgrade, image: postgres:10-alpine move to image: postgres:14-alpine

Did you try to upgrade an existing installation or did you set up a new installation? The migration of an existing database from one Postgres version to another requires extra steps.

> As I try ./artisan migrate, it cause errors as following. 

This looks like the database container is not running. This is probably related to the issue with Docker Compose that you first described.

**Reply by cywhale** [reply]
Thanks your reponse. Yes, I found it's docker compose version problem. I upgrade docker-compose-plugin but /usr/local/bin/docker-compose still old version (1.26.2). So I download newest version from official repo, and now this docker-compose.yaml problem solved.

For the 2nd problem about ./artisan migrate: Ya, I try to upgrade an existing installation. When I finish builded gpu branch, docker ps showed 
```
fb455f7dde1c   postgres:14-alpine   "docker-entrypoint.s…"   9 hours ago   Restarting (1) Less than a second ago                                                                                    biigle-database-1
```
And then I execute ./artisan migrate and got this error.
How can I use some extra steps for this migration? Is there any manual or examples can I follow? Big thanks!

**Reply by mzur** [reply]
The easiest way would be to revert the Docker image in `docker-compose.yaml` to `postgres:10-alpine` again. The database doesn't start because it detects existing database files from an older version of Postgres. If you absolutely want to upgrade the database, you also need to revert to Postgres 10 first, than manually create a database dump, delete the original database, upgrade to the newer Postgres version and recreate the database from the dump. I would recommend to stick with Postgres 10 :wink:

**Reply by cywhale** [reply]
Sure. I revert to postgres 10. Thanks!
